.github/workflows/test_e2e.yml

```diff
@@ -31,22 +44,57 @@ jobs:
           helm: v3.11.2
           kubectl: v1.28.1
           skaffold: v2.3.0
-
-      - name: Run tests
-        env:
-          K3S_CHANNEL: ${{ matrix.k3s }}
-          SCOPE: gha-${{ github.run_id }}-${{ github.run_attempt }}-${{ matrix.k3s }}
-
-          # Domain must be available in the account running the tests. This domain is available in the account
-          # running the public integration tests.
-          CERT_DOMAIN: hc-integrations-test.de
+      - name: install k3sup
         run: |
           curl -sLS https://get.k3sup.dev | sh
 
-          trap "hack/dev-down.sh" EXIT
+      - name: Setup test environment
+        run: |
+          mkdir debug-logs
           source <(hack/dev-up.sh)
+          
+          # make exported env variables available to following jobs
+          echo "KUBECONFIG=$KUBECONFIG" >> "$GITHUB_ENV"
+          echo "SKAFFOLD_DEFAULT_REPO=$SKAFFOLD_DEFAULT_REPO" >> "$GITHUB_ENV"
 
+      - name: Build and Deploy HCCM
+        run: |
           skaffold build --tag="e2e-${GITHUB_RUN_ID}-${GITHUB_RUN_NUMBER}"
           tag=$(skaffold build --tag="e2e-${GITHUB_RUN_ID}-${GITHUB_RUN_NUMBER}" --quiet --output="{{ (index .Builds 0).Tag }}")
           skaffold deploy --images=hetznercloud/hcloud-cloud-controller-manager=$tag
+
+      - name: Run tests
+        run: |
           go test ./tests/e2e -tags e2e -v -timeout 60m
+
+      - name: Download logs & events
+        if: always()
+        continue-on-error: true
+        run: |
+          kubectl logs \
+            --namespace kube-system \
+            --selector app.kubernetes.io/name=hcloud-cloud-controller-manager \
+            --all-containers \
+            --prefix=true \
+            --tail=-1 \
+            > debug-logs/hccm.log
```

Comment:
Should we print the logs in the CI in case of failure ? For small errors one might be able to spot the mistake without having to download the artifacts ?

Reply:
Sounds good, do you think we should log everything or just the hccm logs?

Reply:
I'd go with the most useful one, but not both. HCCM logs seem to be the most useful ones?

Reply:
I just put in the HCCM & hcloud API logs for now. There are too many events to do this nicely.

Reply:
I put them into [log groups](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#grouping-log-lines) that should be collapsed by default. The cloud api logs are only logged if the env setup failed.